@asmorkalov I reproduced the JetPack 4.6 build failure on unmodified OpenCV 5.0 sources.

Inputs:

- OpenCV: `40738fb16ceddb5fb3fea747585f7ce6abb0605b` (`5.0.0`)
- opencv_contrib: `755e50675d97db9b7d449d8bd6b09888646f6c6e` (`5.0.0`)
- Source status: `git status --short` was empty for both trees before/after the build; no patches were applied.
- JetPack 4.6 environment: R32.7.6, CUDA 10.2, GCC/G++ 8.4.0, CMake 3.26.3, cuDNN 8.2.1.
- Build shape mirrored our JetPack CUDA/DNN OpenCV build: static `opencv_world`, `WITH_CUDA=ON`, `WITH_CUDNN=ON`, `WITH_CUBLAS=ON`, `OPENCV_DNN_CUDA=ON`, `CMAKE_CUDA_ARCHITECTURES=72`, and contrib CUDA modules in `BUILD_LIST`.

The JetPack 4.6 build fails in the first contrib `cudaarithm` NVCC batch:

```text
<opencv>/modules/core/include/opencv2/core/cuda.inl.hpp:430:19: error: no matching function for call to 'swap(cv::cuda::GpuMatND&, cv::cuda::GpuMatND&)'
     std::swap(*this, m);
                   ^
/usr/include/c++/8/bits/move.h:182:1: note: candidate: 'template<class _Tp> typename std::enable_if<std::__and_<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> >::value>::type std::swap(_Tp&, _Tp&)'
     swap(_Tp& __a, _Tp& __b)
 ^   
/usr/include/c++/8/bits/move.h:182:1: note:   template argument deduction/substitution failed:
/usr/include/c++/8/bits/move.h: In substitution of 'template<class _Tp> typename std::enable_if<std::__and_<std::__not_<std::__is_tuple_like<_Tp> >, std::is_move_constructible<_Tp>, std::is_move_assignable<_Tp> >::value>::type std::swap(_Tp&, _Tp&) [with _Tp = cv::cuda::GpuMatND]':
<opencv>/modules/core/include/opencv2/core/cuda.inl.hpp:430:19:   required from here
/usr/include/c++/8/bits/move.h:182:1: error: no type named 'type' in 'struct std::enable_if<false, void>'
CMake Error at cuda_compile_1_generated_add_mat.cu.o.Release.cmake:280 (message):
  Error generating file
  <build>/modules/world/CMakeFiles/cuda_compile_1.dir/__/__/__/opencv_contrib/modules/cudaarithm/src/cuda/./cuda_compile_1_generated_add_mat.cu.o

make: *** [opencv_world] Error 2
```

I also checked JetPack 5.1 with the same clean source commits and equivalent CUDA/DNN build shape (`CMAKE_CUDA_ARCHITECTURES=72;87`). Environment was R35.6.3, CUDA 11.4, GCC/G++ 9.4.0, CMake 3.26.3, cuDNN 8.6.0. That build completed successfully:

```text
[100%] Linking CXX static library ../../lib/libopencv_world.a
[100%] Built target opencv_world
```

--- a/modules/core/include/opencv2/core/cuda.hpp
+++ b/modules/core/include/opencv2/core/cuda.hpp
@@ -500,16 +500,8 @@ public:
     GpuMatND(const GpuMatND&) = default;
     GpuMatND& operator=(const GpuMatND&) = default;
 
-#if defined(__GNUC__) && __GNUC__ < 5
-    // error: function '...' defaulted on its first declaration with an exception-specification
-    // that differs from the implicit declaration '...'
-
     GpuMatND(GpuMatND&&) = default;
     GpuMatND& operator=(GpuMatND&&) = default;
-#else
-    GpuMatND(GpuMatND&&) noexcept = default;
-    GpuMatND& operator=(GpuMatND&&) noexcept = default;
-#endif
 
     void upload(InputArray src);
     void upload(InputArray src, Stream& stream);

